lib/fhir_models/bootstrap/definitions.rb

```diff
@@ -216,21 +216,27 @@ def self.get_codes(uri)
     def self.get_display(uri, code)
       return nil if uri.nil? || code.nil?
       load_expansions
-      valuesets = @@expansions.select { |x| x['compose']['include'].detect {|i| i['system'] == uri } }
-      valuesets += @@valuesets.select { |x| x['url'] == uri }
+      valuesets = @@expansions.select { |ex| ex['compose']['include'].detect { |i| i['system'] == uri } }
+      valuesets += @@valuesets.select { |vs| vs['url'] == uri }
       code_hash = nil
       valuesets.each do |valueset|
-        if !valueset['expansion'].nil? && !valueset['expansion']['contains'].nil?
-          code_hash = valueset['expansion']['contains'].select { |x| x['system'] == uri && x['code'] == code}.first
-        elsif !valueset['compose'].nil? && !valueset['compose']['include'].nil?
-          included_systems = valueset['compose']['include'].map { |x| x['system'] }.uniq
-          systems = @@valuesets.select { |x| x['resourceType'] == 'CodeSystem' && included_systems.include?(x['url']) }
-          systems.each do |x|
-            code_hash = x['concept'].select { |con| con['url'] == uri && con['code'] == code }.first if x['concept']
+        if valueset['expansion'] && valueset['expansion']['contains']
+          # This currently only matches 'expansions', not 'valuesets'
+          code_hash = valueset['expansion']['contains'].detect { |contained| contained['system'] == uri && contained['code'] == code }
+        elsif valueset['compose'] && valueset['compose']['include']
+          # I'm not sure this branch ever matches, see comment below.
```

Comment:
```
"compose": {
          "lockedDate": "2012-06-13",
          "include": [
            {
              "system": "http://loinc.org",
              "version": "2.36",
              "concept": [
                {
                  "code": "14647-2",
                  "display": "Cholesterol [Moles/Volume]"
                },
                {
                  "code": "2093-3",
                  "display": "Cholesterol [Mass/Volume]"
                },
                {
                  "code": "35200-5",
                  "display": "Cholesterol [Mass Or Moles/Volume]"
                },
                {
                  "code": "9342-7",
                  "display": "Cholesterol [Percentile]"
                }
              ]
            }
          ]
}
```

The intention of this logic branch was to find valuesets composed of codes from a system, then go to the canonical valueset (e.g. LOINC)  and look up that code.